src/wallet/rpcwallet.cpp

```diff
@@ -3133,9 +3133,13 @@ UniValue z_getnewaddress(const UniValue& params, bool fHelp)
         addrType = params[0].get_str();
     }
 
+    bool allowSapling = Params().NetworkIDString() == "regtest" || (
```

Comment:
What about z_importkey and z_exportkey? Do they need to be gated or not?

Reply:
`z_importkey` is gated because `DecodePaymentAddress` has been modified to remove Sapling support on mainnet and gate it on testnet. `z_exportkey` is handled because we have done the same removal/gating to `z_getnewaddress sapling`, and there is no other way to import a Sapling key into the wallet.